Detectors/TPC/simulation/include/TPCSimulation/Point.h

```diff
@@ -34,24 +45,139 @@ class ElementalHit {
 // and can be stored as element of a TClonesArray into a branch
 class LinkableHitGroup : public o2::BaseHit {
 public:
-  LinkableHitGroup() : mHits() {}
-
-  LinkableHitGroup(int trackID) : mHits() {
+  LinkableHitGroup() :
+#ifdef HIT_AOS
+  mHits()
+#else
+  mHitsX_v(),
+  mHitsY_v(),
+  mHitsZ_v(),
+  mHitsT_v(),
+  mHitsE_v(),
+  mHitsX(nullptr),
+  mHitsY(nullptr),
+  mHitsZ(nullptr),
+  mHitsT(nullptr),
+  mHitsE(nullptr),
+  mSize(0)
+#endif
+    {
+    }
+
+  LinkableHitGroup(int trackID) :
+#ifdef HIT_AOS
+  mHits()
+#else
+  mHitsX_v(),
+  mHitsY_v(),
+  mHitsZ_v(),
+  mHitsT_v(),
+  mHitsE_v(),
+  mHitsX(nullptr),
+  mHitsY(nullptr),
+  mHitsZ(nullptr),
+  mHitsT(nullptr),
+  mHitsE(nullptr),
+  mSize(0)
+#endif
+  {
     SetTrackID(trackID);
   }
 
   ~LinkableHitGroup() override = default;
 
-  void addHit(float x, float y, float z, float time, float e) {
+  void addHit(float x, float y, float z, float time, short e) {
+#ifdef HIT_AOS
     mHits.emplace_back(x,y,z,time,e);
+#else
+    mHitsX_v.emplace_back(x);
+    mHitsY_v.emplace_back(y);
+    mHitsZ_v.emplace_back(z);
+    mHitsT_v.emplace_back(time);
+    mHitsE_v.emplace_back(e);
+    mSize=mHitsX_v.size();
+    mHitsX=&mHitsX_v[0];
+    mHitsY=&mHitsY_v[0];
+    mHitsZ=&mHitsZ_v[0];
+    mHitsT=&mHitsT_v[0];
+    mHitsE=&mHitsE_v[0];
```

Comment:
single vectors for structure of arrays storage. This leads to a "manual" splitting of the container inside the TClonesArray and to a 10% better compression in the ROOT file. (Unfortunately, ROOT is not able to split a std::vector<Hits> inside a TClonesArray itself.

The HIT_AOS option is just to go back to the array of structure storage.